### DISCUSSION TITLE
UNABLE TO BUILD THE TOOLKIT EXTENSION - Have anyone faced this issue? Error: failed to run custom build command for `pgx-pg-sys v0.2.4`

### DISCUSSION THREAD

**Discussion by arunkumar790** [timescale/timescaledb-toolkit#314] [General]
Hi All, I am new to timescaleDB and exploring it, got below error while Installing Toolkit on self-hosted TimescaleDB at step4 (Use Cargo to complete installation)
I couldn't find any solution over net, it would be great if someone help. Thanks
---------------------------------------------------------------
Compiling pgx-pg-sys v0.2.4
error: failed to run custom build command for `pgx-pg-sys v0.2.4`Caused by:
  process didn't exit successfully: `/xxx/xxx/timescaledb-toolkit/target/release/build/pgx-pg-sys-392f32ddaeb2fd57/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=PGX_PG_SYS_SKIP_BINDING_REWRITE
  cargo:rerun-if-changed=/home/arun/.pgx/config.toml
  cargo:rerun-if-changed=include/pg10.h
  cargo:rerun-if-changed=include/pg11.h
  cargo:rerun-if-changed=include/pg12.h
  cargo:rerun-if-changed=include/pg13.h
  cargo:rerun-if-changed=include/pg14.h
  cargo:rerun-if-changed=cshim/pgx-cshim.c
  cargo:rerun-if-changed=cshim/Makefile  --- stderr
  manifest_dir=/xx/xx/.cargo/registry/src/github.com-1ecc6299db9ec823/pgx-pg-sys-0.2.4
  shim_src=/xx/xx/.cargo/registry/src/github.com-1ecc6299db9ec823/pgx-pg-sys-0.2.4/cshim
  shim_dst=/xx/xx/timescaledb-toolkit/target/release/build/pgx-pg-sys-f999c02874ecd61b/out/cshim
  Generating bindings for pg10
  Generating bindings for pg13
  Generating bindings for pg12
  Generating bindings for pg14
  Generating bindings for pg11
  /usr/include/stdio.h:33:10: fatal error: 'stddef.h' file not found
  /usr/include/stdio.h:33:10: fatal error: 'stddef.h' file not found, err: true
  thread '<unnamed>' panicked at 'Unable to generate bindings for pg12: ()', /home/arun/.cargo/registry/src/github.com-1ecc6299db9ec823/pgx-pg-sys-0.2.4/build.rs:499:13
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  /home/arun/.cargo/registry/src/github.com-1ecc6299db9ec823/pgx-pg-sys-0.2.4/include/pg10.h:1:10: fatal error: 'stdbool.h' file not found
  /home/arun/.cargo/registry/src/github.com-1ecc6299db9ec823/pgx-pg-sys-0.2.4/include/pg10.h:1:10: fatal error: 'stdbool.h' file not found, err: true
  /usr/include/stdio.h:33:10: fatal error: 'stddef.h' file not found
  /usr/include/stdio.h:33:10: fatal error: 'stddef.h' file not found, err: true
  thread '<unnamed>' panicked at 'Unable to generate bindings for pg10: ()', /home/arun/.cargo/registry/src/github.com-1ecc6299db9ec823/pgx-pg-sys-0.2.4/build.rs:499:13
  thread '<unnamed>' panicked at 'Unable to generate bindings for pg13: ()', /home/arun/.cargo/registry/src/github.com-1ecc6299db9ec823/pgx-pg-sys-0.2.4/build.rs:499:13
  /usr/include/stdio.h:33:10: fatal error: 'stddef.h' file not found
  /usr/include/stdio.h:33:10: fatal error: 'stddef.h' file not found, err: true
  thread '<unnamed>' panicked at 'Unable to generate bindings for pg11: ()', /home/arun/.cargo/registry/src/github.com-1ecc6299db9ec823/pgx-pg-sys-0.2.4/build.rs:499:13
  /usr/include/stdio.h:33:10: fatal error: 'stddef.h' file not found
  /usr/include/stdio.h:33:10: fatal error: 'stddef.h' file not found, err: true
  thread '<unnamed>' panicked at 'Unable to generate bindings for pg14: ()', /home/arun/.cargo/registry/src/github.com-1ecc6299db9ec823/pgx-pg-sys-0.2.4/build.rs:499:13
      [error] failed to build extension
------------------------------------------------------------------------------------------------------------------------------------------

**Comment by epgts** [comment]
Hi, Arun! It looks like your build system lacks standard C headers such as `stdbool.h` and `stddef.h`. How to get those depends on your operating system, which I don't think you specified. E.g., on Ubuntu 20.04.3, you need to install the linux-libc-dev and libstdc++-9-dev packages.

Our Readme actually suggests running this: `sudo apt-get install make gcc pkg-config clang postgresql-server-dev-13 libssl-dev`.

**Reply by arunkumar790** [reply]
Thanks a lot. I understood docker is much better way and successful with "timescale/timescaledb-ha:pg14-latest"

**Reply by flixman** [reply]
@epgts excuse me for addressing you directly, but tangentially related to this: Yesterday I opened an issue with respect to building the toolkit extension for pg14, as it is giving an error (seems) due to the use of muslc in alpine. Is this the place where I can get help? might you know where else can I look for help, otherwise? Thank you very much for your time! :-)
